### ISSUE TITLE
Filename length limitation

### ISSUE THREAD

**Issue by awoods** [UW-Madison-Library/ocfl-java#4]
The testing and surfacing of this limitation came through the script at the bottom run against a Fedora 6 repository.
The issue, which may potentially require no additional action beyond documenting in the README what the limitation is for file lengths on Ubuntu, is that files with names with 323 characters fails when persisting to OCFL. I have not narrowed down the exact limitation length.
* System
```
$ uname -a
Linux silver 5.3.0-42-generic #34-Ubuntu SMP Fri Feb 28 05:49:40 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 19.10
Release:	19.10
Codename:	eoan
```

Testing script
```bash
clear;x=0; r="http://localhost:8080/rest/"; while true; do x=$(($x + 1));echo $x; r=`curl -XPOST $r`; echo "r = $r"; done 
```
[Script output](https://gist.github.com/awoods/4810c8d4d71e1185079b2de53425f743)

**Comment by pwinckles** [comment]
Cases like this are handled by configuring the repository to enforce path restraints:

https://github.com/UW-Madison-Library/ocfl-java/blob/master/ocfl-java-core/src/main/java/edu/wisc/library/ocfl/core/path/constraint/DefaultContentPathConstraints.java#L72

For example:

```
var repo = new OcflRepositoryBuilder()
        .layoutConfig(DefaultLayoutConfig.nTupleHashConfig())
        .contentPathConstraintProcessor(DefaultContentPathConstraints.unix())
        .storage(FileSystemOcflStorage.builder().repositoryRoot(repoDir).build())
        .workDir(workDir)
        .build();

```

**Comment by pwinckles** [comment]
I should probably rename that property to simply be `contentPathConstraints`.

**Comment by awoods** [comment]
What is the result of configuring the repository with the `DefaultContentPathConstraints.unix()`?

**Comment by pwinckles** [comment]
If you want to support long logical paths you need to define a custom PathSanitizer:

https://github.com/UW-Madison-Library/ocfl-java/blob/master/ocfl-java-core/src/main/java/edu/wisc/library/ocfl/core/path/sanitize/PathSanitizer.java

**Comment by pwinckles** [comment]
In this case it rejects logical paths up front that cannot be safely mapped directly to content paths. PathSanitizers provide a way to transform logical paths into safe content paths, but it's currently BYO.

**Comment by pwinckles** [comment]
I think this is resolved by #5. Currently, the only limitation that ocfl-java has on the length of ocfl object ids is based on the method used to map ids to object root directory paths. If the object id is used as the encapsulation directory, then object ids cannot be longer than 255 characters.
